GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -1,73 +1,42 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-using System.Numerics;
+﻿using System.Linq;
+using GameServerCore;
 using GameServerCore.Domain.GameObjects;
 using GameServerCore.Enums;
-using LeagueSandbox.GameServer.GameObjects.Stats;
 
 namespace LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI
 {
     public class Minion : ObjAiBase, IMinion
     {
-        /// <summary>
-        /// Const waypoints that define the minion's route
-        /// </summary>
-        protected List<Vector2> _mainWaypoints;
-        protected int _curMainWaypoint;
-        public MinionSpawnPosition SpawnPosition { get; }
-        public MinionSpawnType MinionSpawnType { get; }
+        public string Name { get; }
+        public IObjAiBase Owner { get; } // We'll probably want to change this in the future
         protected bool _aiPaused;
 
         public Minion(
-            Game game,
-            MinionSpawnType spawnType,
-            MinionSpawnPosition position,
-            List<Vector2> mainWaypoints,
+            IGame game,
+            IObjAiBase owner,
+            float x,
+            float y,
+            string model,
+            string name,
+            int visionRadius = 0,
             uint netId = 0
-        ) : base(game, "", new Stats.Stats(), 40, 0, 0, 1100, netId)
+        ) : base((Game)game, model, new Stats.Stats(), 40, x, y, visionRadius, netId)
```

Comment:
If you use an actual class and not the interface (Minion and not IMinion) then there is no problem to use the actual Game class.

If there is where is it?